### PR TITLE
chore(workspace): Change default keybinding for Create Scratch Note

### PR DESCRIPTION
**Pull request by hikchoi**
# chore(workspace): Change default keybinding for Create Scratch Note
This PR:
- Changes default keybinding for `Create Scratch Note` from `shift+cmd/ctrl+s` to `cmd/ctrl+k s`
- Add a temporary feature toast during startup to notify the change, and a fail-safe backup toast in the command itself
  - This is only shown once ever, and will be tracked with `FeatureShowcaseDisplayed`

# Pull Request Checklist
### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [~] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [~] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [x] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [~] [Write Tests](dev.process.qa.test) 
- [x] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [~] Common cases tested
- [~] 1-2 Edge cases tested
- [~] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

### Extended
- [~] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), an example is included in the [test workspace](dev.ref.test-workspace)
- CSS
  - [~] display is correct for following dimensions
    - [~] sm: screen ≥ 576px, eg. iphonex, (375x812)
    - [~] lg: screen ≥ 992px
    - [~] xxl: screen ≥ 1600px eg. mac (1600x900)
  - [~] display is correct for following browsers (across the various dimensions)
    - [~] safari
    - [~] firefox
    - [~] chrome

## Docs
- [x] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [~] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

**Comment by hikchoi** [comment]
docs: https://github.com/dendronhq/dendron-site/pull/642

**Comment by hikchoi** [comment]
related: https://github.com/dendronhq/dendron/issues/2200

**Review by jonathanyeung — APPROVED** [review]

File: packages/plugin-core/src/showcase/CreateScratchNoteKeybindingTip.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    return true;
+  }
+
+  get showcaseEntry(): ShowcaseEntry {
+    return ShowcaseEntry.CreateScratchNoteKeybindingTip;
+  }
+
+  getDisplayMessage(_displayLocation: DisplayLocation): string {
+    return `Keyboard shortcut for "Dendron: Create Scratch Note" has changed. If you wish to keep the original, please follow the instructions.`;
```
Comment (kevinslin) [inline packages/plugin-core/src/showcase/CreateScratchNoteKeybindingTip.ts:28]: nitpick: would change the wording to

> If you wish to keep the original, please follow the instructions online

File: packages/plugin-core/src/_extension.ts [hunk 2]
```diff
@@ -353,7 +354,14 @@ export async function _activate(
         const ONE_MINUTE_IN_MS = 60_000;
         setTimeout(() => {
           const showcase = new FeatureShowcaseToaster();
-          showcase.showToast();
+          // Temporarily show the new toast instead of the rest.
+          // for subsequent sessions this will not be shown as it already has been shown.
+          const hasShown = showcase.showSpecificToast(
```
Comment (kevinslin) [inline packages/plugin-core/src/_extension.ts:359]: can we add to the show criteria that we only show this to users who did not just install vscode?

so as part of this change, add a `firstInstallVersion` as part of the install? if `firstInstallVersion` is not undefined or less than `0.112.0`, then show this message

**Review by kevinslin — CHANGES_REQUESTED** [review]

File: packages/plugin-core/src/_extension.ts (hunk 2 above)
Comment (hikchoi) [reply packages/plugin-core/src/_extension.ts:359]: > if firstInstallVersion is not undefined or less than 0.112.0, then show this message

Do you mean if `firstInstallVersion` is _undefined_ or less than 0.112.0? unless we go and backfill them based on the `firstInstall` time one by one, all existing users will have them undefined.

Comment (kevinslin) [reply packages/plugin-core/src/_extension.ts]: the former

**Review by kevinslin — APPROVED** [review]
